CRM/Core/BAO/CustomField.php

```diff
@@ -272,15 +272,21 @@ public static function create(&$params) {
       $params['attributes'] = 'rows=4, cols=60';
     }
 
-    $customField = new CRM_Core_DAO_CustomField();
-    $customField->copyValues($params);
-    $customField->is_required = CRM_Utils_Array::value('is_required', $params, FALSE);
-    $customField->is_searchable = CRM_Utils_Array::value('is_searchable', $params, FALSE);
-    $customField->in_selector = CRM_Utils_Array::value('in_selector', $params, FALSE);
-    $customField->is_search_range = CRM_Utils_Array::value('is_search_range', $params, FALSE);
-    //CRM-15792 - Custom field gets disabled if is_active not set
-    $customField->is_active = CRM_Utils_Array::value('is_active', $params, TRUE);
-    $customField->is_view = CRM_Utils_Array::value('is_view', $params, FALSE);
+    if ($op == 'edit') {
+      $customField = CRM_Core_DAO_CustomField::findById($params['id']);
```

Comment:
I don't get this - I understand that the params should not be applied by code in edit mode - but I don't think we need to worry about loading them so the DB doesn't apply them.

In general I think the preferred way to set defaults is at the api level. Unfortunately historically some form calls to the BAO::create rely on them being set in the BAO - but when done in the BAO the approach is generally to add to the $params in other BAO create functions - eg.

if ($op === 'create') {
$params = array_merge(['is_required' => 0, 'is_searchable' => 0...], $params);
}

But, let's do a quick audit of where create is called & see how much we really are relying on this anti-pattern of setting defaults in the BAO::Create anyway

Reply:
yeah, you are right :) I updated the title and the diff